crates/fluvio-stream-dispatcher/src/metadata/k8.rs

```diff
@@ -0,0 +1,321 @@
+                                changes.push(LSUpdate::Mod(new_kv_value));
+                            }
+                            Err(err) => match err {
+                                K8ConvertError::Skip(obj) => {
+                                    debug!("skipping: {}", obj.metadata.name);
+                                }
+                                _ => {
+                                    error!("converting {} {:#?}", S::LABEL, err);
+                                }
+                            },
+                        }
+                    }
+                    K8Watch::MODIFIED(k8_obj) => {
+                        trace!("{} MODIFIED: {:#?}", S::LABEL, k8_obj);
+                        match S::convert_from_k8(k8_obj, multi_namespace_context) {
+                            Ok(updated_kv_value) => {
+                                debug!(
+                                    "K8: Watch Update {}:{:?}",
+                                    S::LABEL,
+                                    updated_kv_value.key()
+                                );
+                                changes.push(LSUpdate::Mod(updated_kv_value));
+                            }
+                            Err(err) => match err {
+                                K8ConvertError::Skip(obj) => {
+                                    debug!("skipping: {}", obj.metadata.name);
+                                }
+                                _ => {
+                                    error!("converting {} {:#?}", S::LABEL, err);
+                                }
+                            },
+                        }
+                    }
+                    K8Watch::DELETED(k8_obj) => {
+                        trace!("{} DELETE: {:#?}", S::LABEL, k8_obj);
+                        let meta: Result<
+                            MetadataStoreObject<S, K8MetaItem>,
+                            K8ConvertError<S::K8Spec>,
+                        > = S::convert_from_k8(k8_obj, multi_namespace_context);
+                        match meta {
+                            Ok(kv_value) => {
+                                debug!("K8: Watch Delete {}:{:?}", S::LABEL, kv_value.key());
+                                changes.push(LSUpdate::Delete(kv_value.key_owned()));
+                            }
+                            Err(err) => match err {
+                                K8ConvertError::Skip(obj) => {
+                                    debug!("skipping: {}", obj.metadata.name);
+                                }
+                                _ => {
+                                    error!("converting {} {:#?}", S::LABEL, err);
+                                }
+                            },
+                        }
+                    }
+                }
+            }
+
+            Ok(changes)
+        });
+        mapped.boxed()
+    }
+}
+
```

Comment:
Need integration tests are needed

Reply:
Integration tests require k8 API available for them. This might take a lot of effort, but since we do have integration tests in CI, this code is covered.

I added unit-tests with `MemoryClient`. Not all code is covered since `MemoryClient` does not implement all methods that `K8Client` does.

Reply:
> I meant something like this: https://github.com/infinyon/k8-api/blob/master/src/k8-client/tests/service.rs. Need to ensure K8 client maintains same behavior.

But those tests expect that k8 API is running, if you just run `cargo test` they fail with "services: error trying to connect: Connection refused (os error 61)". I don't see much value in such tests.

Reply:
Upon reviewing MemoryClient tests, test seems to be sufficient now.  Can add more tests as warranted